### PR TITLE
Dao8

### PR DESCRIPTION
**Pull request by antoniukivan**

### CODE REVIEW

File: src/main/webapp/WEB.INF/views/cars/create.jsp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/cars/create">
+    Enter car model: <input type="text" name="model">
+    Enter car manufacturer id: <input type="number" name="manufacturerId">
+
+    <button type="submit">Create</button>
+</form>
+</body>
+</html>
```
Comment (yurwing) [inline src/main/webapp/WEB.INF/views/cars/create.jsp]: add empty line in the end of the file

**Review by yurwing — APPROVED** [review]

File: src/main/webapp/WEB.INF/views/cars/drivers/add.jsp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+</head>
+<body>
+<h1>Please, enter data</h1>
+
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/cars/drivers/add">
+    Enter car id: <input type="number" name="car_id">
+    Enter driver id: <input type="number" name="driver_id">
+
```
Comment (Ilya-brinchuk) [inline src/main/webapp/WEB.INF/views/cars/drivers/add.jsp]: ты можешь добавить во всех инпутах  required="" и не надо будет проверять на NullPointerEx

File: src/main/webapp/WEB.INF/views/manufacturers/create.jsp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/manufacturers/create">
+    Enter manufacturer name: <input type="text" name="name">
+    Enter manufacturer country: <input type="text" name="country">
+
+    <button type="submit">Create</button>
+</form>
+</body>
+</html>
```
Comment (Ilya-brinchuk) [inline src/main/webapp/WEB.INF/views/manufacturers/create.jsp]: добавь пустую строку

**Review by Ilya-brinchuk — CHANGES_REQUESTED** [review]

**Review by Ilya-brinchuk — APPROVED** [review]

File: src/main/java/mate/academy/controller/AddDriverToCarController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+public class AddDriverToCarController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB.INF/views/cars/drivers/add.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/AddDriverToCarController.java]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/cars/drivers/add.jsp").forward(req, resp);
```
change the name of this folder

File: src/main/java/mate/academy/controller/AddDriverToCarController.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            Long driverId = Long.valueOf(req.getParameter("driver_id"));
+            Long carId = Long.valueOf(req.getParameter("car_id"));
+            Car car = carService.get(carId);
+            Driver driver = driverService.get(driverId);
+            carService.addDriverToCar(driver, car);
+        } catch (NumberFormatException | DataProcessingException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/AddDriverToCarController.java]: catching RuntimeExceptions is bad practice - let's remove this try-catch and assume that we will always receive valid ids here

File: src/main/webapp/WEB.INF/views/cars/drivers/add.jsp [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    <title>Add driver to the car</title>
+</head>
+<body>
+<h1>Please, enter data</h1>
+
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/cars/drivers/add">
+    Enter car id: <input required="" type="number" name="car_id">
+    Enter driver id: <input required="" type="number" name="driver_id">
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB.INF/views/cars/drivers/add.jsp]: you already check on the JSP that the input should be number, so there is no need to catch NumberFormatException

File: src/main/java/mate/academy/controller/CreateCarController.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        try {
+            Long manufacturerId = Long.valueOf(req.getParameter("manufacturerId"));
+            Manufacturer manufacturer = manufacturerService.get(manufacturerId);
+            carService.create(new Car(model, manufacturer));
+        } catch (NumberFormatException | DataProcessingException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/CreateCarController.java]: same here - let's remove this try-catch block

File: src/main/java/mate/academy/controller/CreateCarController.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB.INF/views/cars/create.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        try {
+            Long manufacturerId = Long.valueOf(req.getParameter("manufacturerId"));
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/CreateCarController.java]:
```suggestion
            Long manufacturerId = Long.valueOf(req.getParameter("manufacturer_id"));
```
use the same approach everywhere

File: src/main/java/mate/academy/controller/CreateDriverController.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB.INF/views/drivers/create.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String licenseNumber = req.getParameter("licenseNumber");
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/CreateDriverController.java]:
```suggestion
        String licenseNumber = req.getParameter("license_number");
```

File: src/main/java/mate/academy/controller/CreateDriverController.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        if (name.isEmpty() || licenseNumber.isEmpty()) {
+            req.setAttribute("message", "Please enter valid data");
+            req.getRequestDispatcher("/WEB.INF/views/drivers/create.jsp").forward(req, resp);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/CreateDriverController.java]: let's remove this

File: src/main/webapp/WEB.INF/views/drivers/create.jsp [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    <title>Create driver</title>
+</head>
+<body>
+<h1>Please, enter information about new driver</h1>
+
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/drivers/create">
+    Enter driver name: <input required="" type="text" name="name">
+    Enter driver license number: <input required="" type="text" name="licenseNumber">
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB.INF/views/drivers/create.jsp]:
```suggestion
    Enter driver name: <input type="text" name="name" required>
    Enter driver license number: <input type="text" name="license_number" required>
```

File: src/main/java/mate/academy/controller/CreateManufacturerController.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB.INF/views/manufacturers/create.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String country = req.getParameter("country");
+        if (name.isEmpty() || country.isEmpty()) {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/CreateManufacturerController.java]: same for the manufacturer

File: src/main/webapp/WEB.INF/views/manufacturers/create.jsp [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    <title>Create manufacturer</title>
+</head>
+<body>
+<h1>Please, enter information about new manufacturer</h1>
+
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/manufacturers/create">
+    Enter manufacturer name: <input required="" type="text" name="name">
+    Enter manufacturer country: <input required="" type="text" name="country">
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB.INF/views/manufacturers/create.jsp]:
```suggestion
    Enter manufacturer name: <input type="text" name="name" required>
    Enter manufacturer country: <input type="text" name="country" required>
```

File: src/main/webapp/WEB.INF/views/manufacturers/create.jsp [hunk 14]
```diff
@@ -0,0 +1,18 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Create manufacturer</title>
+</head>
+<body>
+<h1>Please, enter information about new manufacturer</h1>
+
+<h4 style="color: red">${message}</h4>
+
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB.INF/views/manufacturers/create.jsp]:
```suggestion
```
we will be able to remove this

File: src/main/webapp/WEB.INF/views/cars/create.jsp [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+    <title>Create car</title>
+</head>
+<body>
+<h1>Please, enter information about new car</h1>
+
+<h4 style="color: red">${message}</h4>
+
+<form method="post" action="${pageContext.request.contextPath}/cars/create">
+    Enter car model: <input required="" type="text" name="model">
+    Enter car manufacturer id: <input required="" type="number" name="manufacturerId">
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB.INF/views/cars/create.jsp]: same comment here as for drivers and manufacturers

File: src/main/webapp/web.xml [hunk 16]
```diff
@@ -6,11 +6,71 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allDrivers</servlet-name>
+        <servlet-class>mate.academy.controller.GetAllDriversController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allDrivers</servlet-name>
+        <url-pattern>/drivers/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/drivers/</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controller/DeleteCarController.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private final CarService carService
+            = (CarService) injector.getInstance(CarService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long carId = Long.valueOf(req.getParameter("id"));
+        carService.delete(carId);
+        resp.sendRedirect(req.getContextPath() + "/cars/all");
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/DeleteCarController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/cars");
```
do you have this URL `/cars/all` mapped in web.xml?

File: src/main/java/mate/academy/controller/DeleteDriverController.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long driverId = Long.valueOf(req.getParameter("id"));
+        driverService.delete(driverId);
+        resp.sendRedirect(req.getContextPath() + "/drivers/all");
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/DeleteDriverController.java]: same

File: src/main/java/mate/academy/controller/DeleteManufacturerController.java [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private final ManufacturerService manufacturerService
+            = (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long manufacturerId = Long.valueOf(req.getParameter("id"));
+        manufacturerService.delete(manufacturerId);
+        resp.sendRedirect(req.getContextPath() + "/manufacturers/all");
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controller/DeleteManufacturerController.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**